tools/azure-sdk-tools/devtools_testutils/azure_recorded_testcase.py

```diff
@@ -261,3 +189,105 @@ def generate_sas(self, *args, **kwargs):
         sas_func_pos_args = args[1:]
         token = sas_func(*sas_func_pos_args, **kwargs)
         return token
+
+def get_credential(is_live=True, **kwargs):
+    tenant_id = os.environ.get("AZURE_TENANT_ID", getattr(os.environ, "TENANT_ID", None))
+    client_id = os.environ.get("AZURE_CLIENT_ID", getattr(os.environ, "CLIENT_ID", None))
+    secret = os.environ.get("AZURE_CLIENT_SECRET", getattr(os.environ, "CLIENT_SECRET", None))
+
+    use_pwsh = os.environ.get("AZURE_TEST_USE_PWSH_AUTH", "false")
+    use_cli = os.environ.get("AZURE_TEST_USE_CLI_AUTH", "false")
+    use_vscode = os.environ.get("AZURE_TEST_USE_VSCODE_AUTH", "false")
+    use_azd = os.environ.get("AZURE_TEST_USE_AZD_AUTH", "false")
+    is_async = kwargs.pop("is_async", False)
+
+    # Return live credentials only in live mode
+    if is_live:
+        # User-based authentication through Azure PowerShell, if requested
+        if use_pwsh.lower() == "true":
+            _LOGGER.info(
+                "Environment variable AZURE_TEST_USE_PWSH_AUTH set to 'true'. Using AzurePowerShellCredential."
+            )
+            from azure.identity import AzurePowerShellCredential
+
+            if is_async:
+                from azure.identity.aio import AzurePowerShellCredential
+            return AzurePowerShellCredential()
```

Comment:
While we're making changes to `get_credential`, what do you think about passing `**kwargs` into credential constructors? I'm thinking about use cases like ACR's, where they specify a particular `authority`. I can also understand handling this in a separate PR.

Reply:
Good point.

Let's add it.